# KoNLPy 설치
### JPype도 설치해줘야 한다. (공식 홈페이지 참고)

In [1]:
from konlpy.tag import Kkma
from konlpy.utils import pprint
kkma = Kkma()
print(kkma.sentences(u'설치가 잘 되었는지 Test1을 합니다.'))

['설치가 잘 되었는지 Test1 을 합니다.']


In [2]:
print(kkma.nouns(u'설치가 잘 되었는지 Test1을 합니다.'))

['설치', '1']


In [3]:
print(kkma.pos(u'설치가 잘 되었는지 Test1을 합니다.'))

[('설치', 'NNG'), ('가', 'JKS'), ('잘', 'MAG'), ('되', 'VV'), ('었', 'EPT'), ('는지', 'ECS'), ('Test', 'OL'), ('1', 'NR'), ('을', 'JKO'), ('하', 'VV'), ('ㅂ니다', 'EFN'), ('.', 'SF')]


# Kaggle API
Kaggle 공식 깃허브 참고 (https://github.com/Kaggle/kaggle-api)
account에서 API 발급 받고 깃허브 문서에서 API파일 위치 대로 옮겨 놓는다

In [4]:
!pip show kaggle

Name: kaggle
Version: 1.5.3
Summary: Kaggle API
Home-page: https://github.com/Kaggle/kaggle-api
Author: Kaggle
Author-email: support@kaggle.com
License: Apache 2.0
Location: c:\python36\virtualenv\venv01\lib\site-packages
Requires: certifi, tqdm, python-slugify, requests, six, python-dateutil, urllib3
Required-by: 


In [7]:
!kaggle competitions list -h

usage: kaggle competitions list [-h] [--group GROUP] [--category CATEGORY]
                                [--sort-by SORT_BY] [-p PAGE] [-s SEARCH] [-v]

optional arguments:
  -h, --help            show this help message and exit
  --group GROUP         Search for competitions in a specific group. Default is 'general'. Valid options are 'general', 'entered', and 'inClass'
  --category CATEGORY   Search for competitions of a specific category. Default is 'all'. Valid options are 'all', 'featured', 'research', 'recruitment', 'gettingStarted', 'masters', and 'playground'
  --sort-by SORT_BY     Sort list results. Default is 'latestDeadline'. Valid options are 'grouped', 'prize', 'earliestDeadline', 'latestDeadline', 'numberOfTeams', and 'recentlyCreated'
  -p PAGE, --page PAGE  Page number for results paging. Page size is 20 by default
  -s SEARCH, --search SEARCH
                        Term(s) to search for
  -v, --csv             Print results in CSV format (if not set print in table 

In [8]:
# nlp에 해당되는 경진대회 찾기
!kaggle competitions list -s nlp

ref                    deadline             category            reward  teamCount  userHasEntered  
---------------------  -------------------  ---------------  ---------  ---------  --------------  
word2vec-nlp-tutorial  2015-06-30 23:59:00  Getting Started  Knowledge        578           False  


In [9]:
# nlp에 해당되는 경진대회 데이터셋 보기
!kaggle competitions files -c word2vec-nlp-tutorial

name                         size  creationDate         
--------------------------  -----  -------------------  
sampleSubmission.csv        276KB  2014-07-21 22:24:56  
unlabeledTrainData.tsv.zip   26MB  2014-08-05 23:31:46  
testData.tsv.zip             13MB  2014-08-05 23:31:52  
labeledTrainData.tsv.zip     13MB  2014-08-05 23:32:11  


In [11]:
# titanic에 해당되는 경진대회 데이터셋 보기
!kaggle competitions files -c titanic

name                   size  creationDate         
---------------------  ----  -------------------  
train.csv              60KB  2013-06-28 13:40:25  
test.csv               28KB  2013-06-28 13:40:24  
gender_submission.csv   3KB  2017-02-01 01:49:18  


In [13]:
# 타이타닉 경진대회 파일 다운로드
!kaggle competitions download -c titanic

403 - Forbidden


    403-Frobidden 이 뜨는 경우 Rule에 대한 동의를 하지 않아 다운 받을 수 없는 것이다.
    https://www.kaggle.com/c/<competition-name>/rules
    다운 받을 데이터셋이나 competition의 Data 또는 Rule 섹션 하단의 Rule에 대해 동의한다

In [16]:
# 타이타닉 경진대회 파일 다운로드
!kaggle competitions download -c titanic


503 - Service Unavailable



  0%|          | 0.00/59.8k [00:00<?, ?B/s]
100%|##########| 59.8k/59.8k [00:00<00:00, 718kB/s]


### Submission이나 Dataset 같은 옵션은 공식 문서나 강의 영상 참고

In [1]:
!kaggle datasets list -s survey

No datasets found
